src/erasure-code/jerasure/jerasure_init.cc

```diff
@@ -3,7 +3,7 @@
 /*
  * Ceph distributed storage system
  *
- * Copyright (C) 2014 Cloudwatt <libre.licensing@cloudwatt.com>
```

Comment:
I realize just now that the initialization of jerasure has no impact on how the plugin exposes the w parameter. Yesterday when we discussed it I argued against having common code for both jerasure and shec but I was wrong. Sorry for the noise.

to answer your question, the variables in shec and jerasure are different and the initialization done on shec won't be used by jerasure.